### PR TITLE
StudentProfileCreate: values should be trimmed automatically #2389

### LINKED ISSUE #2389: studentProfileCreate: values should be trimmed automatically
It is not nice to reject values because of leading/trailing spaces.

### PR DESCRIPTION
**Pull request by rcyq**
Fixes #2389

### CODE REVIEW

**Comment by rcyq** [comment]
@damithc Can I merge?

File: src/main/java/teammates/ui/controller/StudentProfileEditSaveAction.java [hunk 1]
```diff
@@ -50,8 +50,35 @@ private StudentProfileAttributes extractProfileData() {
...
-        
+        preprocessParameters(editedProfile);
+        validatePostParameters(editedProfile);        
+
         return editedProfile;
     }
+
+    private void preprocessParameters(StudentProfileAttributes studentProfile) {
+        if(studentProfile.shortName != null){
+            studentProfile.shortName = studentProfile.shortName.trim();;
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/StudentProfileEditSaveAction.java]: double semicolon

File: src/main/java/teammates/ui/controller/StudentProfileEditSaveAction.java [hunk 2]
```diff
@@ -50,8 +50,35 @@ private StudentProfileAttributes extractProfileData() {
...
+        }
+       
+        if(studentProfile.email != null){
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/StudentProfileEditSaveAction.java]: create a `trimIfNotNull` method in `StringHelper`?

File: src/test/java/teammates/test/cases/common/FieldValidatorTest.java [hunk 3]
```diff
@@ -256,7 +256,7 @@ public void testGetValidityInfoForAllowedName() {
         String untrimmedValue = " abc ";
         assertEquals("invalid: untrimmed", 
                 String.format(WHITESPACE_ONLY_OR_EXTRA_WHITESPACE_ERROR_MESSAGE, typicalFieldName),
-                validator.getValidityInfoForSizeCappedNonEmptyString(
+                validator.getValidityInfoForAllowedName(
                         typicalFieldName, 
                         maxLength, 
                         untrimmedValue));
```
Comment (damithc) [inline src/test/java/teammates/test/cases/common/FieldValidatorTest.java]: remove this file

File: src/test/java/teammates/test/cases/ui/StudentProfileEditSaveActionTest.java [hunk 4]
```diff
@@ -125,12 +125,30 @@ private void verifyLogMessage(AccountAttributes student,
...
         StudentProfileAttributes spa = new StudentProfileAttributes();
-        spa.shortName = submissionParams[1];
-        spa.email = submissionParams[3];
-        spa.institute = submissionParams[5];
-        spa.nationality = submissionParams[7];
-        spa.gender = submissionParams[9];
-        spa.moreInfo = submissionParams[11];
+        
+        if(submissionParams[1] != null){
+            spa.shortName = submissionParams[1].trim();;
```
Comment (damithc) [inline src/test/java/teammates/test/cases/ui/StudentProfileEditSaveActionTest.java]: so many double semi-colons

**Comment by damithc** [comment]
Some minor changes requested.

**Comment by rcyq** [comment]
Changes made

**Pull request merged**